components/dialog/test/dialog.visual-diff.js

```diff
@@ -47,22 +47,32 @@ describe('d2l-dialog', () => {
 					});
 
 					it('opened', async function() {
-						if (info.category === 'short-narrow') return; // TODO: remove this; skipping for now due to flaking
-						await helper.open(page, '#dialog');
-						await visualDiff.screenshotAndCompare(page, this.test.fullTitle());
+						return Promise.all([
+							helper.open(page, '#dialog'),
+							page.waitForSelector('#bottom', { visible: true })
+						]).then(() => {
+							return visualDiff.screenshotAndCompare(page, this.test.fullTitle());
+						});
 					});
 
 					it('opened-wide', async function() {
-						await page.$eval('#wideContainer', wideContainer => wideContainer.style.width = '1500px');
-						await helper.open(page, '#dialog');
-						await visualDiff.screenshotAndCompare(page, this.test.fullTitle());
-						await page.$eval('#wideContainer', wideContainer => wideContainer.style.width = 'auto');
+						return Promise.all([
```

Comment:
Is the `Promise.all` needed?  Does this work?

```javascript
await page.$eval('#wideContainer', wideContainer => wideContainer.style.width = '1500px');
await helper.open(page, '#dialog');
await page.waitForSelector('#bottom', { visible: true });

await visualDiff.screenshotAndCompare(page, this.test.fullTitle());
await page.$eval('#wideContainer', wideContainer => wideContainer.style.width = 'auto');
```
If we _do_ need the `Promise.all` for some reason, I think it would also be fine to write it like this:
```javascript
await Promise.all([
	page.$eval('#wideContainer', wideContainer => wideContainer.style.width = '1500px'),
	helper.open(page, '#dialog'),
	page.waitForSelector('#bottom', { visible: true })
]);
await visualDiff.screenshotAndCompare(page, this.test.fullTitle());
await page.$eval('#wideContainer', wideContainer => wideContainer.style.width = 'auto');
```
Not a big deal.  The `then` statements just aren't as tidy.

In the end, the key requirement here is that we need to use `page.waitForSelector('#bottom', { visible: true })` for some reason eh?

Reply:
Worth noting that `Promise.all` does not run in sequence, but in parallel. So there's no guarantee that it'll wait for the first thing to execute before doing the second... which I think is what you need?

Reply:
So I think for `opened-wide` it wasn't flaky before and we can go back to how it was with the `await`s though I'll need to test that out (it's hard to know because something will be totally fine running locally but really flaky in the PR). The `Promise.all` seems to be necessary for the `opened` and `rtl` tests (I believe when I had that was when there was consistently no flake) but I'll try to restructure if possible.